ob1k-core/src/main/java/com/outbrain/ob1k/client/DoubleDispatchStrategy.java

```diff
@@ -0,0 +1,26 @@
+package com.outbrain.ob1k.client;
+
+import com.outbrain.ob1k.concurrent.Try;
+
+/**
+ * An interface which enables to define a custom double dispatch strategy for HttpClient.
+ * This mechanism is required in order to ensure the target provider will not use the same target for the two invocation of the double dispatch
+ *
+ *  @author lifey
+ */
+public interface DoubleDispatchStrategy {
+  /**
+   * Implement this method to determine how long should the double dispatch delay be.
```

Comment:
gj for the docs, but maybe a javadoc that specifies only the api without "implement this" would look a bit better :)
same for all methods